File: 151004/Sakovsky/src/entities/Author.ts
import { Column, Entity, PrimaryGeneratedColumn, Table } from "typeorm";

@Entity({name: 'tbl_authors'})
// @Table()
export class Author {
    @PrimaryGeneratedColumn()
    id: number;

    @Column({unique: true})
    login: string;

    @Column()
    password: string;

    @Column()
    firstname: string;

    @Column()
    lastname: string;
}

File: 151002/Rusakovich/NewsDistributedSystem/publisher/src/main/java/by/rusakovich/publisher/author/model/Author.java
package by.rusakovich.publisher.author.model;

import by.rusakovich.publisher.generics.model.IEntity;
import by.rusakovich.publisher.news.model.News;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.List;


@Getter
@Setter
@Entity
@Table(name = "tbl_author")
public class Author implements IEntity<Long> {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, length = 64)
    private String login;

    @Column(length = 128)
    private String password;

    @Column(length = 64)
    private String firstname;

    @Column(length = 64)
    private String lastname;

    @OneToMany(mappedBy = "author")
    private List<News> news;
}
